### PR TITLE
Update backend dependencies

### PR DESCRIPTION
**Pull request by I-Al-Istannen**
## Organizational changes:
* *All* dependencies and plugin versions are now configured via the properties.
* Some exclusions and default settings have been defined for the maven versions plugin.

#### Exemplary maven version plugin workflow
1. `mvn versions:display-property-updates` -- Think about which ones you need
2. `mvn versions:update-properties` -- And be careful nothing you didn't want was changed
3. `mvn versions:commit` to accept these changes or `mvn versions:revert` to undo them

## Major dependency version changes in this PR:
1. JOOQ uses java 8 time datatypes now. I configured it to rewrite SQLite's TIMESTAMP to java.time.Instant
2. Mockito was updated from 1.X to 3.X, but I think not much has changed

### CODE REVIEW

**Comment by Garmelon** [comment]
The updating workflow could be worth an entry in the wiki.

File: backend/pom.xml [hunk 1]
```diff
@@ -20,31 +20,71 @@
...
+    <maven-project-info-reports-plugin.version>3.1.1</maven-project-info-reports-plugin.version>
   </properties>
 
 
   <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <version>3.0.0-M3</version>
```
Comment (Garmelon) [inline backend/pom.xml:47]: If you made variables for all other versions, why not for this one?

**Review by Garmelon — APPROVED** [review]

File: backend/pom.xml (hunk 1 above)
Comment (I-Al-Istannen) [reply backend/pom.xml:47]: Human error. Let me rebase the PR

**Review by Garmelon — APPROVED** [review]

**Review by kwerber — APPROVED** [review]

**Comment by Garmelon** [comment]
Because #60 was created and merged before this, it didn't extract its version numbers into properties. This is fixed with this commit.

**Review by kwerber — APPROVED** [review]

**Pull request merged**